crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs

```diff
@@ -215,7 +213,16 @@ impl Simulator {
                     }
                 },
             );
-        l2_blocks_with_no_da_blocks.chain(da_blocks).collect()
+        let da_block_ranges = da_blocks.into_iter().map(|maybe_recorded_blocks| {
+            maybe_recorded_blocks.map(|list| {
+                let heights_iter = list.iter().map(|(height, _, _)| *height);
+                let min = heights_iter.clone().min().unwrap();
+                let max = heights_iter.max().unwrap();
```

Comment:
Nit:
We could potentially get `min` and `max` with a single pass over the list. But since it's only in the simulator, not the actual production code, we can leave this as is for clarity.

Reply:
Can be left as is.